src/recipe.h

```diff
@@ -50,6 +51,26 @@ struct recipe_proficiency {
     void deserialize( JsonIn &jsin );
 };
 
+struct book_recipe_data {
+    int skill_req = -1;
+    bool use_alt_name = false;
+    std::string alt_name = "";
+    bool hidden = false;
+
+    book_recipe_data( JsonArray arr ) {
+        int arr_size = arr.size();
+        if( arr_size > 1 ) {
```

Comment:
Instead of using a confusing array, consider using an object and read the values using keys.